app/src/main/java/io/manycore/robotcoder/demo/MainActivity.kt

```diff
@@ -0,0 +1,27 @@
+package io.manycore.robotcoder.demo
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import java.io.File
+
+class MainActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        val storageDirectory = getExternalFilesDir("your_path")?.absolutePath
```

Comment:
getExternalFilesDir is accessing the filesystem. This means a bad performance impact.

I recommend to move it to a coroutine.